src/main/java/io/lyuda/jcards/Dice.java

```diff
@@ -0,0 +1,52 @@
+package io.lyuda.jcards;
+
+import java.security.SecureRandom;
+
+/**
+ * The {@code Dice} class represents a dice with an arbitrary amount of sides, defined in the constructor.
+ * This class contains methods to access the number of sides, as well as rolling the dice to generate a random number.
+ *
+ * @author KonnosBaz
+*/
+public class Dice {
+
+    /**
+     * The amount of sides on the dice.
+     */
+    private int sides;
```

Comment:
Storing the last rolled number and providing a method to retrieve it could be beneficial.

There are games where dice are "held" - they are not rerolled and in such cases, being able to get the last rolled number is necessary.

Also, a "hold" property could be helpful. This property would prevent the die from being rolled. - comes in handy later when multiple dice are involved in games like "Yahtzee" etc.

Reply:
I added a way to store the last roll but I think holding from within the class might be unnecessary. Since the last roll is stored, holding just means not rolling and I think it would be easier to program that into specific games, rather than complicating the class and adding a way to block the roll function.